project_generator/exporters/uvision.py

```diff
@@ -149,12 +150,24 @@ def generate(self, data, env_settings):
         mcu_def_dic = self.definitions.get_mcu_definition(expanded_dic['mcu'])
         self.append_mcu_def(expanded_dic, mcu_def_dic)
 
+        # set default build directory if unset
+        if not 'output_dir' in expanded_dic:
+            expanded_dic['output_dir'] = '.\\build\\' + data['name'] + '\\'
```

Comment:
I did this to exactly reproduce the original behaviour (backslashes were baked into the template file). AFAIK µvision is windows only in any case.